StripeCore/StripeCore/Source/Attestation/StripeAttest.swift

```diff
@@ -26,10 +26,51 @@ import UIKit
         } catch {
             let errorAnalytic = ErrorAnalytic(event: .assertionFailed, error: error)
             STPAnalyticsClient.sharedClient.log(analytic: errorAnalytic, apiClient: apiClient)
-            throw error
+            if apiClient.isTestmode {
+                // In testmode, we can provide a test assertion even if the real assertion fails
+                return testmodeAssertion
+            } else {
+                throw error
+            }
+        }
+    }
+
+    /// Determines if the current device is able to sign requests.
+    /// If the device has not attested previously, we will create a key and attest it.
+    /// If `true`, the device is ready for attestation. If `false`, attestation is not possible.
+    @_spi(STP) public func prepareAttestation() async -> Bool {
+        do {
+            if !Self.successfullyAttested {
+                // We haven't attested yet, so do that first.
+                try await self.attest()
+            }
+
+            return Self.successfullyAttested
+        } catch {
+            return false
         }
     }
 
+    /// Inform StripeAttest of an error received in response to an assertion.
+    /// The key may be reset if needed.
+    @_spi(STP) public func receivedAssertionError(_ error: Error) {
+        // If the backend responds with this specific error, we should reset the key.
+        if let error = error as? StripeCore.StripeError,
+           case let .apiError(apiError) = error,
+           apiError.code == "invalid_parameter",
+           apiError.type == .invalidRequestError,
+           apiError.param == nil {
+            let resetKeyAnalytic = ErrorAnalytic(event: .resetKeyForAssertionError, error: error)
+            STPAnalyticsClient.sharedClient.log(analytic: resetKeyAnalytic, apiClient: apiClient)
+            resetKey()
+        }
```

Comment:
Seems odd that we only log invalid_parameter errors.  Shouldn't we log others?

Reply:
StripeAttest isn't interested in other errors — this is the error from the endpoint being asserted against (might be a `Link` endpoint, might be `PaymentIntents/create`, etc), so we only handle it if it's one specific type of error.